src/vsr/client.zig

```diff
@@ -117,6 +123,10 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
                 replica_count: u8,
                 message_pool: *MessagePool,
                 message_bus_options: MessageBus.Options,
+                eviction_callback: ?*const fn (
+                    client: *Self,
+                    eviction: *const Message.Eviction,
+                ) void = null,
```

Comment:
Could you write some docs what should be done after callback? I see that basically every method has `assert(!evicted)`, so presumably the client shuold not be used afterwards. Should the caller then call deinit?